alacritty_terminal/src/term/mod.rs

```diff
@@ -281,6 +281,52 @@ impl<'a, C> RenderableCellsIter<'a, C> {
             cursor_style,
         }
     }
+
+    /// Check selection state of a cell.
+    fn is_selected(&self, point: Point) -> bool {
+        let selection = match self.selection {
+            Some(selection) => selection,
+            None => return false,
+        };
+
+        // Point itself is selected
+        if selection.contains(point.col, point.line) {
+            return true;
+        }
+
+        let num_cols = self.grid.num_cols().0;
+        let cell = self.grid[&point];
+
+        // Check if wide char's spacers are selected
+        if cell.flags.contains(Flags::WIDE_CHAR) {
+            let prevprev = point.sub(num_cols, 2, false);
+            let prev = point.sub(num_cols, 1, false);
+            let next = point.add(num_cols, 1, false);
+
+            // Check trailing spacer
+            return selection.contains(next.col, next.line)
+                // Check line-wrapping, leading spacer
+                || (self.grid[&prev].flags.contains(Flags::WIDE_CHAR_SPACER)
+                    && !self.grid[&prevprev].flags.contains(Flags::WIDE_CHAR)
+                    && selection.contains(prev.col, prev.line));
+        }
+
+        // Check if spacer's wide char is selected
+        if cell.flags.contains(Flags::WIDE_CHAR_SPACER) {
+            let prev = point.sub(num_cols, 1, false);
+
+            if self.grid[&prev].flags.contains(Flags::WIDE_CHAR) {
+                // Check previous cell for trailing spacer
+                return self.is_selected(prev);
+            } else {
+                // Check next cell for line-wrapping, leading spacer
+                let next = point.add(num_cols, 1, false);
+                return self.is_selected(next);
+            }
+        }
+
+        false
```

Comment:
I feel like you can just do the following instead of this strange `return` logic.

This is pseudo code.

```rust

if cell.flags... {
	let prev = point.sub
	let point = if self.grid... {
		&prev
	} else {
		&next
	}
	
	self.is_selected(point)
} else {
	false
}

```

Reply:
I don't think the `let point = if { ... } else { }` benefits us here, but the rest of this looks good.